libsolidity/formal/SMTEncoder.cpp

```diff
@@ -304,11 +307,49 @@ void SMTEncoder::endVisit(Block const& _block)
 
 bool SMTEncoder::visit(InlineAssembly const& _inlineAsm)
 {
-	m_errorReporter.warning(
-		7737_error,
-		_inlineAsm.location(),
-		"Assertion checker does not support inline assembly."
-	);
+	struct AssignedExternalsCollector : public yul::ASTWalker
+	{
+		using ExternalReferences = map<yul::Identifier const*, InlineAssemblyAnnotation::ExternalIdentifierInfo>;
+		AssignedExternalsCollector(InlineAssembly const& _inlineAsm) : externalReferences(_inlineAsm.annotation().externalReferences)
+		{
+			this->operator()(_inlineAsm.operations());
+		}
+
+		ExternalReferences const& externalReferences;
+		set<VariableDeclaration const*> assignedVars;
+
+		using yul::ASTWalker::operator();
+		void operator()(yul::Assignment const& _assignment) {
+			auto const& vars = _assignment.variableNames;
+			for (auto const& identifier: vars)
+				if (auto it = externalReferences.find(&identifier); it != externalReferences.end())
+					if (auto declaration = dynamic_cast<VariableDeclaration const*>(it->second.declaration))
+						assignedVars.insert(declaration);
+		}
+	};
+
+	auto assignedVars = AssignedExternalsCollector(_inlineAsm).assignedVars;
+
+	yul::SideEffectsCollector sideEffectsCollector(_inlineAsm.dialect(), _inlineAsm.operations());
+	bool writesToMemory = sideEffectsCollector.invalidatesMemory();
+	bool writesToStorage = sideEffectsCollector.invalidatesStorage();
+
+	if (writesToMemory)
+		resetMemoryVariables();
+	if (writesToStorage)
+		resetStorageVariables();
+	for (auto const* var: assignedVars)
+		if (var)
+		{
+			solAssert(var->isLocalVariable(), "Non-local variable assigned in inlined assembly");
+			m_context.resetVariable(*var);
+		}
+
+//	m_errorReporter.warning(
+//		7737_error,
+//		_inlineAsm.location(),
+//		"Assertion checker does not support inline assembly."
+//	);
```

Comment:
Do we want to keep this warning, remove it, or replace with something like "Results might be imprecise because of over-approximation caused by inline assembly"?

Reply:
We can add the message you suggested when we see we need to reset something, and can omit the message when nothing had to be reset

Reply:
I have changed the message of the warning.
Since there will always be something over-approximated (if the assembly block is doing anything), I think it is OK to display it every time.